Add missing prototypes

When building with `-Wstrict-prototypes`, the following warnings are generated:

```
In file included from src/hashfunctions.c:14:
In file included from src/hashfunctions.h:17:
src/datastructures.h:23:28: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
typedef Obj (*GVarFuncType)(/*arguments*/);
                           ^
                                         void
In file included from src/datastructures.c:12:
src/datastructures.h:23:28: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
typedef Obj (*GVarFuncType)(/*arguments*/);
                           ^
                                         void
In file included from src/avl.c:6:
In file included from src/avl.h:9:
src/datastructures.h:23:28: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
typedef Obj (*GVarFuncType)(/*arguments*/);
                           ^
                                         void
In file included from src/binaryheap.c:19:
In file included from src/binaryheap.h:15:
src/datastructures.h:23:28: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
typedef Obj (*GVarFuncType)(/*arguments*/);
                           ^
                                         void
In file included from src/hashmap.c:10:
In file included from src/hashmap.h:14:
src/datastructures.h:23:28: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
typedef Obj (*GVarFuncType)(/*arguments*/);
                           ^
                                         void
In file included from src/pairingheap.c:16:
In file included from src/pairingheap.h:18:
src/datastructures.h:23:28: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
typedef Obj (*GVarFuncType)(/*arguments*/);
                           ^
                                         void
In file included from src/skiplist.c:6:
In file included from src/skiplist.h:9:
src/datastructures.h:23:28: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
typedef Obj (*GVarFuncType)(/*arguments*/);
                           ^
                                         void
In file included from src/uf.c:6:
In file included from src/uf.h:9:
src/datastructures.h:23:28: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
typedef Obj (*GVarFuncType)(/*arguments*/);
                           ^
                                         void
```

This isn't too serious _right now_, but the next C standard will likely make this an error. We're trying to get ahead of the problem: https://wiki.gentoo.org/wiki/Modern_C_porting

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -27,8 +27,6 @@ jobs:
           - master
           - stable-4.13
           - stable-4.12
-          - stable-4.11
-          - stable-4.10
         ABI: ['']
         include:
           - gap-branch: master

--- a/PackageInfo.g
+++ b/PackageInfo.g
@@ -115,7 +115,7 @@ PackageDoc := [ rec(
 ##  Are there restrictions on the operating system for this package? Or does
 ##  the package need other packages to be available?
 Dependencies := rec(
-  GAP := ">= 4.10",
+  GAP := ">= 4.12",
   NeededOtherPackages := [["GAPDoc", "1.5"]],
   SuggestedOtherPackages := [],
   # OtherPackagesLoadedInAdvance := [],

--- a/src/avl.c
+++ b/src/avl.c
@@ -12,7 +12,7 @@ enum {
     AVL_FLAGS = 4
 };
 
-Obj DS_AVL_FIND(Obj self, Obj tree, Obj val, Obj less)
+static Obj FuncDS_AVL_FIND(Obj self, Obj tree, Obj val, Obj less)
 {
     const UInt lmask = 0x4;
     const UInt rmask = 0x8;
@@ -54,7 +54,7 @@ Obj DS_AVL_FIND(Obj self, Obj tree, Obj val, Obj less)
 
 
 static Obj
-DS_AVL_ADDSET_INNER(Obj self, Obj avl, Obj val, Obj less, Obj trinode)
+FuncDS_AVL_ADDSET_INNER(Obj self, Obj avl, Obj val, Obj less, Obj trinode)
 {
     Obj  child;
     UInt dirn;
@@ -97,7 +97,7 @@ DS_AVL_ADDSET_INNER(Obj self, Obj avl, Obj val, Obj less, Obj trinode)
     }
     else {
         child = ELM_PLIST(avl, i);
-        deeper = DS_AVL_ADDSET_INNER((Obj)0, child, val, less, trinode);
+        deeper = FuncDS_AVL_ADDSET_INNER((Obj)0, child, val, less, trinode);
         if (deeper == INTOBJ_INT(0)) {
             /* we just have to increase the size of subtree */
             SET_ELM_PLIST(avl, AVL_FLAGS, INTOBJ_INT(flags + 0x10));
@@ -126,7 +126,7 @@ DS_AVL_ADDSET_INNER(Obj self, Obj avl, Obj val, Obj less, Obj trinode)
 
 
 
-static Obj DS_AVL_REMSET_INNER(Obj self,
+static Obj FuncDS_AVL_REMSET_INNER(Obj self,
                                 Obj node,
                                 Obj val,
                                 Obj less,
@@ -163,7 +163,7 @@ static Obj DS_AVL_REMSET_INNER(Obj self,
     if (flags & imask) {
         GAP_ASSERT(ELM_PLIST(node, i));
 	child = ELM_PLIST(node, i);
-        ret = DS_AVL_REMSET_INNER(0, child, val, less,
+        ret = FuncDS_AVL_REMSET_INNER(0, child, val, less,
                                    remove_extremal, trinode, remove_this);
         if (ret == Fail)
             return Fail;
@@ -206,11 +206,9 @@ static Obj DS_AVL_REMSET_INNER(Obj self,
 
 
 static StructGVarFunc GVarFuncs[] = {
-    GVARFUNC(DS_AVL_FIND, 3, "avl, val, lessFunc"),
-    GVARFUNC(DS_AVL_ADDSET_INNER, 4, "avl, val, lessFunc, trinode"),
-    GVARFUNC(DS_AVL_REMSET_INNER,
-             6,
-             "avl, val, lessFunc, remove_extremal, trinode, remove_this"),
+    GVAR_FUNC_3ARGS(DS_AVL_FIND, avl, val, lessFunc),
+    GVAR_FUNC_4ARGS(DS_AVL_ADDSET_INNER, avl, val, lessFunc, trinode),
+    GVAR_FUNC_6ARGS(DS_AVL_REMSET_INNER, avl, val, lessFunc, remove_extremal, trinode, remove_this),
     { 0 }
 };
 

--- a/src/binaryheap.c
+++ b/src/binaryheap.c
@@ -99,7 +99,7 @@ static Int DS_BinaryHeap_BubbleDown(Obj data, Obj isLess, Int i)
     return i;
 }
 
-Obj DS_BinaryHeap_Insert(Obj self, Obj heap, Obj elm)
+static Obj FuncDS_BinaryHeap_Insert(Obj self, Obj heap, Obj elm)
 {
     Obj data = DS_BINARYHEAP_DATA(heap);
     Obj isLess = DS_BINARYHEAP_ISLESS(heap);
@@ -118,7 +118,7 @@ Obj DS_BinaryHeap_Insert(Obj self, Obj heap, Obj elm)
     return 0;
 }
 
-Obj DS_BinaryHeap_ReplaceMax(Obj self, Obj heap, Obj elm)
+static Obj FuncDS_BinaryHeap_ReplaceMax(Obj self, Obj heap, Obj elm)
 {
     Obj data = DS_BINARYHEAP_DATA(heap);
     Obj isLess = DS_BINARYHEAP_ISLESS(heap);
@@ -136,8 +136,8 @@ Obj DS_BinaryHeap_ReplaceMax(Obj self, Obj heap, Obj elm)
 }
 
 static StructGVarFunc GVarFuncs[] = {
-    GVARFUNC(DS_BinaryHeap_Insert, 2, "heap, elm"),
-    GVARFUNC(DS_BinaryHeap_ReplaceMax, 2, "heap, elm"),
+    GVAR_FUNC_2ARGS(DS_BinaryHeap_Insert, heap, elm),
+    GVAR_FUNC_2ARGS(DS_BinaryHeap_ReplaceMax, heap, elm),
     { 0 }
 };
 

--- a/src/datastructures.h
+++ b/src/datastructures.h
@@ -13,15 +13,6 @@
 
 #include "compiled.h" // GAP headers
 
-// Helper macro to simplify initialization of StructGVarFunc records
-#define GVARFUNC(name, nparam, params)                                       \
-    {                                                                        \
-        #name, nparam, params, (GVarFuncType) name, __FILE__ ":" #name       \
-    }
-
-// This typedef is used by the GVARFUNC macro.
-typedef Obj (*GVarFuncType)(/*arguments*/);
-
 
 // To improve code separation, each data structure implementation can
 // provide a DatastructuresModule struct similar to GAP's StructInitInfo.

--- a/src/hashfunctions.c
+++ b/src/hashfunctions.c
@@ -65,7 +65,7 @@ Int DataHashFuncForPerm(Obj perm)
     }
 }
 
-Obj DATA_HASH_FUNC_FOR_PERM(Obj self, Obj perm)
+static Obj FuncDATA_HASH_FUNC_FOR_PERM(Obj self, Obj perm)
 {
     if (TNUM_OBJ(perm) != T_PERM2 && TNUM_OBJ(perm) != T_PERM4) {
         ErrorMayQuit("DATA_HASH_FUNC_FOR_PERM: <perm> must be a permutation "
@@ -76,7 +76,7 @@ Obj DATA_HASH_FUNC_FOR_PERM(Obj self, Obj perm)
     return HashValueToObjInt(DataHashFuncForPerm(perm));
 }
 
-Obj DATA_HASH_FUNC_FOR_PPERM(Obj self, Obj pperm)
+static Obj FuncDATA_HASH_FUNC_FOR_PPERM(Obj self, Obj pperm)
 {
     if (!IS_PPERM(pperm)) {
         ErrorMayQuit("DATA_HASH_FUNC_FOR_PPERM: <pperm> must be a "
@@ -87,7 +87,7 @@ Obj DATA_HASH_FUNC_FOR_PPERM(Obj self, Obj pperm)
     return HashValueToObjInt(HashFuncForPPerm(pperm));
 }
 
-Obj DATA_HASH_FUNC_FOR_TRANS(Obj self, Obj trans)
+static Obj FuncDATA_HASH_FUNC_FOR_TRANS(Obj self, Obj trans)
 {
     if (!IS_TRANS(trans)) {
         ErrorMayQuit("DATA_HASH_FUNC_FOR_TRANS: <trans> must be a "
@@ -98,7 +98,7 @@ Obj DATA_HASH_FUNC_FOR_TRANS(Obj self, Obj trans)
     return HashValueToObjInt(HashFuncForTrans(trans));
 }
 
-Obj DATA_HASH_FUNC_FOR_STRING(Obj self, Obj string)
+static Obj FuncDATA_HASH_FUNC_FOR_STRING(Obj self, Obj string)
 {
     if (!IS_STRING(string)) {
         ErrorMayQuit("DATA_HASH_FUNC_FOR_STRING: <string> must be a "
@@ -132,7 +132,7 @@ Int DataHashFuncForInt(Obj i)
     }
 }
 
-Obj DATA_HASH_FUNC_FOR_INT(Obj self, Obj i)
+static Obj FuncDATA_HASH_FUNC_FOR_INT(Obj self, Obj i)
 {
     if (TNUM_OBJ(i) != T_INT && TNUM_OBJ(i) != T_INTPOS &&
         TNUM_OBJ(i) != T_INTNEG) {
@@ -242,13 +242,13 @@ static inline Int BasicRecursiveHash(Obj obj)
     return 0;
 }
 
-Obj DATA_HASH_FUNC_RECURSIVE1(Obj self, Obj obj)
+static Obj DATA_HASH_FUNC_RECURSIVE1(Obj self, Obj obj)
 {
     Int hash = BasicRecursiveHash(obj);
     return HashValueToObjInt(hash);
 }
 
-Obj DATA_HASH_FUNC_RECURSIVE2(Obj self, Obj obj1, Obj obj2)
+static Obj DATA_HASH_FUNC_RECURSIVE2(Obj self, Obj obj1, Obj obj2)
 {
     UInt hash1 = BasicRecursiveHash(obj1);
     UInt hash2 = BasicRecursiveHash(obj2);
@@ -259,7 +259,7 @@ Obj DATA_HASH_FUNC_RECURSIVE2(Obj self, Obj obj1, Obj obj2)
     return HashValueToObjInt(listhash2);
 }
 
-Obj DATA_HASH_FUNC_RECURSIVE3(Obj self, Obj obj1, Obj obj2, Obj obj3)
+static Obj DATA_HASH_FUNC_RECURSIVE3(Obj self, Obj obj1, Obj obj2, Obj obj3)
 {
     Int hash1 = BasicRecursiveHash(obj1);
     Int hash2 = BasicRecursiveHash(obj2);
@@ -272,7 +272,7 @@ Obj DATA_HASH_FUNC_RECURSIVE3(Obj self, Obj obj1, Obj obj2, Obj obj3)
     return HashValueToObjInt(listhash3);
 }
 
-Obj DATA_HASH_FUNC_RECURSIVE4(Obj self, Obj obj1, Obj obj2, Obj obj3, Obj obj4)
+static Obj DATA_HASH_FUNC_RECURSIVE4(Obj self, Obj obj1, Obj obj2, Obj obj3, Obj obj4)
 {
     Int hash1 = BasicRecursiveHash(obj1);
     Int hash2 = BasicRecursiveHash(obj2);
@@ -291,11 +291,11 @@ Obj DATA_HASH_FUNC_RECURSIVE4(Obj self, Obj obj1, Obj obj2, Obj obj3, Obj obj4)
 // Submodule declaration
 //
 static StructGVarFunc GVarFuncs[] = {
-    GVARFUNC(DATA_HASH_FUNC_FOR_STRING, 1, "string"),
-    GVARFUNC(DATA_HASH_FUNC_FOR_TRANS, 1, "trans"),
-    GVARFUNC(DATA_HASH_FUNC_FOR_PPERM, 1, "pperm"),
-    GVARFUNC(DATA_HASH_FUNC_FOR_PERM, 1, "perm"),
-    GVARFUNC(DATA_HASH_FUNC_FOR_INT, 1, "int"),
+    GVAR_FUNC_1ARGS(DATA_HASH_FUNC_FOR_STRING, string),
+    GVAR_FUNC_1ARGS(DATA_HASH_FUNC_FOR_TRANS, trans),
+    GVAR_FUNC_1ARGS(DATA_HASH_FUNC_FOR_PPERM, pperm),
+    GVAR_FUNC_1ARGS(DATA_HASH_FUNC_FOR_PERM, perm),
+    GVAR_FUNC_1ARGS(DATA_HASH_FUNC_FOR_INT, int),
     { 0 }
 };
 

--- a/src/hashmap.c
+++ b/src/hashmap.c
@@ -324,7 +324,7 @@ static void DS_RequireMutable(Obj ht)
 // high-level functions, to be called from GAP
 //
 
-Obj DS_Hash_Create(Obj self, Obj hashfunc, Obj eqfunc, Obj capacity, Obj novalues)
+static Obj FuncDS_Hash_Create(Obj self, Obj hashfunc, Obj eqfunc, Obj capacity, Obj novalues)
 {
     if (TNUM_OBJ(hashfunc) != T_FUNCTION) {
         ErrorQuit("<hashfunc> must be a function (not a %s)",
@@ -379,38 +379,38 @@ Obj DS_Hash_Create(Obj self, Obj hashfunc, Obj eqfunc, Obj capacity, Obj novalue
     return ht;
 }
 
-Obj DS_Hash_Capacity(Obj self, Obj ht)
+static Obj FuncDS_Hash_Capacity(Obj self, Obj ht)
 {
     DS_RequireHashMapOrSet(ht);
     Obj keys = ELM_PLIST(ht, POS_KEYS);
     return INTOBJ_INT(LEN_PLIST(keys));
 }
 
-Obj DS_Hash_Used(Obj self, Obj ht)
+static Obj FuncDS_Hash_Used(Obj self, Obj ht)
 {
     DS_RequireHashMapOrSet(ht);
     return ELM_PLIST(ht, POS_USED);
 }
 
-Obj _DS_Hash_Lookup(Obj self, Obj ht, Obj key)
+static Obj Func_DS_Hash_Lookup(Obj self, Obj ht, Obj key)
 {
     DS_RequireHashMapOrSet(ht);
     return INTOBJ_INT(_DS_Hash_Lookup_MayCreate(ht, key, 0));
 }
 
-Obj _DS_Hash_LookupCreate(Obj self, Obj ht, Obj key)
+static Obj Func_DS_Hash_LookupCreate(Obj self, Obj ht, Obj key)
 {
     DS_RequireHashMapOrSet(ht);
     return INTOBJ_INT(_DS_Hash_Lookup_MayCreate(ht, key, 1));
 }
 
-Obj DS_Hash_Contains(Obj self, Obj ht, Obj key)
+static Obj FuncDS_Hash_Contains(Obj self, Obj ht, Obj key)
 {
     DS_RequireHashMapOrSet(ht);
     return _DS_Hash_Lookup_MayCreate(ht, key, 0) != 0 ? True : False;
 }
 
-Obj DS_Hash_Value(Obj self, Obj ht, Obj key)
+static Obj FuncDS_Hash_Value(Obj self, Obj ht, Obj key)
 {
     DS_RequireHashMap(ht);
     UInt idx = _DS_Hash_Lookup_MayCreate(ht, key, 0);
@@ -420,7 +420,7 @@ Obj DS_Hash_Value(Obj self, Obj ht, Obj key)
     return ELM_PLIST(values, idx);
 }
 
-Obj DS_Hash_Reserve(Obj self, Obj ht, Obj new_capacity)
+static Obj FuncDS_Hash_Reserve(Obj self, Obj ht, Obj new_capacity)
 {
     DS_RequireHashMapOrSet(ht);
     DS_RequireMutable(ht);
@@ -448,14 +448,14 @@ Obj DS_Hash_Reserve(Obj self, Obj ht, Obj new_capacity)
     return 0;
 }
 
-Obj DS_Hash_SetValue(Obj self, Obj ht, Obj key, Obj val)
+static Obj FuncDS_Hash_SetValue(Obj self, Obj ht, Obj key, Obj val)
 {
     DS_RequireHashMap(ht);
     DS_RequireMutable(ht);
     return _DS_Hash_SetOrAccValue(ht, key, val, 0);
 }
 
-Obj DS_Hash_AccumulateValue(Obj self, Obj ht, Obj key, Obj val, Obj accufunc)
+static Obj FuncDS_Hash_AccumulateValue(Obj self, Obj ht, Obj key, Obj val, Obj accufunc)
 {
     DS_RequireHashMap(ht);
     DS_RequireMutable(ht);
@@ -466,7 +466,7 @@ Obj DS_Hash_AccumulateValue(Obj self, Obj ht, Obj key, Obj val, Obj accufunc)
     return _DS_Hash_SetOrAccValue(ht, key, val, accufunc);
 }
 
-Obj DS_Hash_AddSet(Obj self, Obj ht, Obj key)
+static Obj FuncDS_Hash_AddSet(Obj self, Obj ht, Obj key)
 {
     DS_RequireHashSet(ht);
     DS_RequireMutable(ht);
@@ -474,7 +474,7 @@ Obj DS_Hash_AddSet(Obj self, Obj ht, Obj key)
     return 0;
 }
 
-Obj DS_Hash_Delete(Obj self, Obj ht, Obj key)
+static Obj FuncDS_Hash_Delete(Obj self, Obj ht, Obj key)
 {
     DS_RequireHashMapOrSet(ht);
     DS_RequireMutable(ht);
@@ -500,22 +500,22 @@ Obj DS_Hash_Delete(Obj self, Obj ht, Obj key)
 
 
 static StructGVarFunc GVarFuncs[] = {
-    GVARFUNC(DS_Hash_Create, 4, "hashfunc, eqfunc, capacity, novalues"),
+    GVAR_FUNC_4ARGS(DS_Hash_Create, hashfunc, eqfunc, capacity, novalues),
 
-    GVARFUNC(DS_Hash_Capacity, 1, "ht"),
-    GVARFUNC(DS_Hash_Used, 1, "ht"),
+    GVAR_FUNC_1ARGS(DS_Hash_Capacity, ht),
+    GVAR_FUNC_1ARGS(DS_Hash_Used, ht),
 
-    GVARFUNC(_DS_Hash_Lookup, 2, "ht, key"),
-    GVARFUNC(_DS_Hash_LookupCreate, 2, "ht, key"),
-    GVARFUNC(DS_Hash_Contains, 2, "ht, key"),
-    GVARFUNC(DS_Hash_Value, 2, "ht, key"),
+    GVAR_FUNC_2ARGS(_DS_Hash_Lookup, ht, key),
+    GVAR_FUNC_2ARGS(_DS_Hash_LookupCreate, ht, key),
+    GVAR_FUNC_2ARGS(DS_Hash_Contains, ht, key),
+    GVAR_FUNC_2ARGS(DS_Hash_Value, ht, key),
 
-    GVARFUNC(DS_Hash_Reserve, 2, "ht, capacity"),
-    GVARFUNC(DS_Hash_SetValue, 3, "ht, key, val"),
-    GVARFUNC(DS_Hash_AccumulateValue, 4, "ht, key, val, accufunc"),
-    GVARFUNC(DS_Hash_AddSet, 2, "ht, key"),
+    GVAR_FUNC_2ARGS(DS_Hash_Reserve, ht, capacity),
+    GVAR_FUNC_3ARGS(DS_Hash_SetValue, ht, key, val),
+    GVAR_FUNC_4ARGS(DS_Hash_AccumulateValue, ht, key, val, accufunc),
+    GVAR_FUNC_2ARGS(DS_Hash_AddSet, ht, key),
 
-    GVARFUNC(DS_Hash_Delete, 2, "ht, key"),
+    GVAR_FUNC_2ARGS(DS_Hash_Delete, ht, key),
 
     { 0 }
 };

--- a/src/pairingheap.c
+++ b/src/pairingheap.c
@@ -29,7 +29,7 @@ enum {
     NODE_POS_SUBHEAPS,    // list of subheaps
 };
 
-Obj DS_merge_pairs(Obj self, Obj isLess, Obj heaps)
+static Obj FuncDS_merge_pairs(Obj self, Obj isLess, Obj heaps)
 {
     if (!IS_DENSE_PLIST(heaps))
         ErrorQuit("<heaps> is not a dense plist", 0L, 0L);
@@ -95,7 +95,7 @@ Obj DS_merge_pairs(Obj self, Obj isLess, Obj heaps)
 }
 
 static StructGVarFunc GVarFuncs[] = {
-    GVARFUNC(DS_merge_pairs, 2, "isLess, heaps"),
+    GVAR_FUNC_2ARGS(DS_merge_pairs, isLess, heaps),
     { 0 }
 };
 

--- a/src/skiplist.c
+++ b/src/skiplist.c
@@ -6,7 +6,7 @@
 #include "skiplist.h"
 
 
-static Obj DS_Skiplist_RemoveNode(Obj self, Obj lst, Obj nx)
+static Obj FuncDS_Skiplist_RemoveNode(Obj self, Obj lst, Obj nx)
 {
     UInt level, len;
     Obj  node, x;
@@ -35,7 +35,7 @@ static Obj DS_Skiplist_RemoveNode(Obj self, Obj lst, Obj nx)
 }
 
 
-static Obj DS_Skiplist_Scan(Obj self, Obj sl, Obj val, Obj lessFunc)
+static Obj FuncDS_Skiplist_Scan(Obj self, Obj sl, Obj val, Obj lessFunc)
 {
     Obj  ptr;
     UInt level;
@@ -76,8 +76,8 @@ static Obj DS_Skiplist_Scan(Obj self, Obj sl, Obj val, Obj lessFunc)
 
 
 static StructGVarFunc GVarFuncs[] = {
-    GVARFUNC(DS_Skiplist_Scan, 3, "skiplist, val, lessFunc"),
-    GVARFUNC(DS_Skiplist_RemoveNode, 2, "lst, nx"),
+    GVAR_FUNC_3ARGS(DS_Skiplist_Scan, skiplist, val, lessFunc),
+    GVAR_FUNC_2ARGS(DS_Skiplist_RemoveNode, lst, nx),
     { 0 }
 };
 

--- a/src/uf.c
+++ b/src/uf.c
@@ -13,7 +13,7 @@
 
 #define RANKMASK ((1L << RANKBITS) -1)
 
-static Obj DS_UF_FIND(Obj self, Obj xo, Obj data) {
+static Obj FuncDS_UF_FIND(Obj self, Obj xo, Obj data) {
   UInt x = INT_INTOBJ(xo);
   UInt y;
   UInt z;
@@ -39,9 +39,9 @@ static Obj DS_UF_FIND(Obj self, Obj xo, Obj data) {
   }
 }
 
-static Obj DS_UF_UNITE(Obj self, Obj xo, Obj yo, Obj data) {
-  UInt x = INT_INTOBJ(DS_UF_FIND(0, xo, data));
-  UInt y = INT_INTOBJ(DS_UF_FIND(0, yo, data));
+static Obj FuncDS_UF_UNITE(Obj self, Obj xo, Obj yo, Obj data) {
+  UInt x = INT_INTOBJ(FuncDS_UF_FIND(0, xo, data));
+  UInt y = INT_INTOBJ(FuncDS_UF_FIND(0, yo, data));
   UInt rx, ry;
   if (x == y)
     return False;
@@ -61,8 +61,8 @@ static Obj DS_UF_UNITE(Obj self, Obj xo, Obj yo, Obj data) {
 
 
 static StructGVarFunc GVarFuncs[] = {
-  GVARFUNC(DS_UF_FIND, 2, "x, data"),
-  GVARFUNC(DS_UF_UNITE, 3, "x, y, data"),
+    GVAR_FUNC_2ARGS(DS_UF_FIND, x, data),
+    GVAR_FUNC_3ARGS(DS_UF_UNITE, x, y, data),
     { 0 }
 };
 

